Chono/AnnotationSource.cs

```diff
@@ -213,22 +247,101 @@ private void PopulateAnnotations(int bookNum, int chapter)
+						}
+						else if (markerTok.Type == MarkerType.Paragraph)
+							atParaStart = true;
+	                }
+					else if (tok is IUSFMTextToken textTok)
+	                {
+		                var text = textTok.Text;
+		                int start = 0;
+		                var continuer = false;
+
+		                void AddAnnotation(Capture capture, bool openingNestedQuote = false)
+		                {
+			                var cCaptureCharsIncludedInSel = 
+				                (openingNestedQuote ? 0 : capture.Length);
+			                annotations.Add(new Annotation(
+				                new Selection(
+					                text.Substring(start,
+						                capture.Index + cCaptureCharsIncludedInSel - start),
+					                text.Substring(0, start),
+					                text.Substring(capture.Index + cCaptureCharsIncludedInSel),
+					                textTok.VerseRef,
+					                textTok.VerseOffset + start),
+				                kQuoteStylePrefix + currentQuoteLevel));
+		                }
+
+		                foreach (Match match in m_findMarksRegex.Matches(text))
+		                {
+							// We skip the first one, which is always "0".
+			                foreach (var matchGroup in match.SuccessfulMatchGroups().Skip(1))
+			                {
+				                if (matchGroup.Name.StartsWith(kRgxOpenLevelGroupPrefix))
+				                {
+					                var validMatchLevels = matchGroup.Name.Substring(3);
+					                var index = validMatchLevels.IndexOf(currentQuoteLevel.ToString(),
+						                StringComparison.Ordinal);
+					                if (index >= 0)
+					                {
+						                if (currentQuoteLevel > 0)
+						                {
+							                continuer = atParaStart && index > 0;
+											if (!continuer)
+												AddAnnotation(match, true);
+						                }
+
+						                start = match.Index;
+										if (!continuer)
+											currentQuoteLevel++;
+					                }
+					                break;
+				                }
+
+				                if (matchGroup.Name.StartsWith(kRgxCloserGroupPrefix))
+				                {
+					                var validMatchLevels = matchGroup.Name.Substring(3);
+					                if (validMatchLevels.Contains(currentQuoteLevel.ToString()))
+					                {
+						                AddAnnotation(match);
+						                if (--currentQuoteLevel > 0)
+							                start = match.Index + match.Length;
+					                }
+					                break;
+				                }
+
+				                if (matchGroup.Name == kRgxFinalGroup && currentQuoteLevel > 0)
+				                {
+					                AddAnnotation(match);
+					                break;
+				                }
+			                
+							}
+
+			                atParaStart = false;
+		                }
+		                
+		                atParaStart = false;
```

Comment:
Bad mix of spaces and tabs for this method. I don't know if you care, but it looks bad in GitHub. :)